compiler_opt/tools/regalloc_trace/extract_functions_lib.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A library that contains utilities for extracting functions."""
+
+import subprocess
+import os
+import shutil
+import json
+import concurrent.futures
+from collections.abc import Sequence
+
+
+def _get_functions_in_file(bc_file_path: str, llvm_nm_path: str) -> list[str]:
+  """Gets all function names defined in a file.
```

Comment:
it's functions that are defined and survive ElimAvailExternal and other optimizations and make it to the point they'd be available to the linker. Right?

Reply:
Yes. It should be symbols with external/weak linkage.